### ISSUE TITLE
Installation of JavaCV

### ISSUE THREAD

**Issue by DSanghera** [natar-io/PapARt#7]
Hello,

I keep getting this error when running PCConfiguration example: **No library found for org.bytedeco.javacpp.opencv_core**

I downloaded the JavaCV files you provided on the site and copied them into the library folder of processing, but it keeps giving me this error. I'm guessing that I have not installed JavaCV correctly. I'm using Mac, would you be able to describe exactly how to install JavaCV to get it to work with processing in this way?

Thank you so much.

**Comment by DSanghera** [comment]
This is the complete error I get: 

No library found for org.bytedeco.javacpp.opencv_core
No library found for org.bytedeco.javacpp.opencv_core
No library found for org.bytedeco.javacpp.opencv_core
No object for introspection, we use PApplet.
skatolo 2.1.1 infos, comments, questions at https://github.com/potioc/skatolo
Display 69732482
Display 889192645
640 480
java.lang.ArrayIndexOutOfBoundsException: 1
	at processing.awt.PGraphicsJava2D$ImageCache.update(PGraphicsJava2D.java:1768)
	at processing.awt.PGraphicsJava2D.imageImpl(PGraphicsJava2D.java:1604)
	at processing.core.PGraphics.textCharModelImpl(PGraphics.java:5045)
	at processing.core.PGraphics.textCharImpl(PGraphics.java:5015)
	at processing.core.PGraphics.textLineImpl(PGraphics.java:4990)
	at processing.awt.PGraphicsJava2D.textLineImpl(PGraphicsJava2D.java:2080)
	at processing.core.PGraphics.textLineAlignImpl(PGraphics.java:4980)
	at processing.core.PGraphics.text(PGraphics.java:4653)
	at processing.core.PGraphics.text(PGraphics.java:4604)
	at fr.inria.skatolo.ControlFont.draw(ControlFont.java:314)
	at fr.inria.skatolo.gui.Label.draw(Label.java:199)
	at fr.inria.skatolo.gui.Label$SinglelineLabel.draw(Label.java:521)
	at fr.inria.skatolo.gui.Label.draw(Label.java:163)
	at fr.inria.skatolo.gui.controllers.Toggle$ToggleView.display(Toggle.java:266)
	at fr.inria.skatolo.gui.controllers.Toggle$ToggleView.display(Toggle.java:255)
	at fr.inria.skatolo.gui.controllers.Toggle.draw(Toggle.java:93)
	at fr.inria.skatolo.gui.group.ControllerGroup.drawControllers(ControllerGroup.java:588)
	at fr.inria.skatolo.gui.group.ControllerGroup.draw(ControllerGroup.java:569)
	at fr.inria.skatolo.gui.group.ControllerGroup.drawControllers(ControllerGroup.java:588)
	at fr.inria.skatolo.gui.group.ControllerGroup.draw(ControllerGroup.java:569)
	at fr.inria.skatolo.gui.ControlWindow.draw(ControlWindow.java:714)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at processing.core.PApplet$RegisteredMethods.handle(PApplet.java:1428)
	at processing.core.PApplet$RegisteredMethods.handle(PApplet.java:1421)
	at processing.core.PApplet.handleMethods(PApplet.java:1615)
	at processing.core.PApplet.handleDraw(PApplet.java:2451)
	at processing.awt.PSurfaceAWT$12.callDraw(PSurfaceAWT.java:1547)
	at processing.core.PSurfaceNone$AnimationThread.run(PSurfaceNone.java:316)

**Comment by poqudrof** [comment]
Is it at PCConfiguration at startup ? 

It is a quite weird error indeed. I guess the error comes from a Font error in Processing.

**Comment by poqudrof** [comment]
Go in  your ui.pde file : https://github.com/poqudrof/Papart-examples/blob/master/papart-examples/calibration/PCConfiguration/ui.pde 

Try to replace the  `arial` font on line 29 to another font name available on your system. Maybe there is no error when the font is not available in the system.

**Comment by Mameng** [comment]
Hello, 

I almost have the same problem as I'm running the PCConfiguration.pde : 
**No library found for org.bytedeco.javacpp.opencv_core.**

I guess there's something missing in my Processing's libraries installation but I'm wondering if I need to add the following java interface somewhere in my Processing libraries : **https://github.com/bytedeco/javacv**

I also replaced the line 29 in the "UI" class but nothing changed. 

More precisely, when I'm running the .pde there's no "big" file error except the previous one (No library…). But as soon as I'm selecting the Processing camera option, I'm getting the following error message.

Thank you for your help.


sept. 30, 2017 6:44:10 PM fr.inria.skatolo.events.ControlBroadcaster printMethodError
GRAVE: An error occured while forwarding a Controller event, please check your code at testCameraButton
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at fr.inria.skatolo.events.ControlBroadcaster.invokeMethod(ControlBroadcaster.java:315)
	at fr.inria.skatolo.events.ControlBroadcaster.callTarget(ControlBroadcaster.java:288)
	at fr.inria.skatolo.events.ControlBroadcaster.broadcast(ControlBroadcaster.java:272)
	at fr.inria.skatolo.gui.Controller.broadcast(Controller.java:1536)
	at fr.inria.skatolo.gui.controllers.Button.setValue(Button.java:153)
	at fr.inria.skatolo.gui.controllers.Button.activate(Button.java:135)
	at fr.inria.skatolo.gui.controllers.Button.mouseReleased(Button.java:107)
	at fr.inria.skatolo.gui.Controller.setMouseReleased(Controller.java:658)
	at fr.inria.skatolo.gui.Controller.setMousePressed(Controller.java:635)
	at fr.inria.skatolo.gui.group.ControllerGroup.setMousePressed(ControllerGroup.java:786)
	at fr.inria.skatolo.gui.ControlWindow.mouseReleasedEvent(ControlWindow.java:795)
	at fr.inria.skatolo.gui.ControlWindow.updatePointerEvents(ControlWindow.java:641)
	at fr.inria.skatolo.gui.ControlWindow.draw(ControlWindow.java:675)
	at sun.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at processing.core.PApplet$RegisteredMethods.handle(PApplet.java:1427)
	at processing.core.PApplet$RegisteredMethods.handle(PApplet.java:1420)
	at processing.core.PApplet.handleMethods(PApplet.java:1614)
	at processing.core.PApplet.handleDraw(PApplet.java:2450)
	at processing.awt.PSurfaceAWT$12.callDraw(PSurfaceAWT.java:1557)
	at processing.core.PSurfaceNone$AnimationThread.run(PSurfaceNone.java:316)
Caused by: java.lang.NullPointerException
	at processing.video.Capture.addResFromSourceMacHack(Unknown Source)
	at processing.video.Capture.addResFromSource(Unknown Source)
	at processing.video.Capture.checkResIsValid(Unknown Source)
	at processing.video.Capture.start(Unknown Source)
	at fr.inria.papart.procam.camera.CameraProcessing.start(CameraProcessing.java:48)
	at PCConfiguration$TestView.testCamera(PCConfiguration.java:454)
	at PCConfiguration.testCameraButton(PCConfiguration.java:147)
	... 26 more

**Comment by Mameng** [comment]
Re bonjour, 

@poqudrof j'écris en français ce sera plus précis ;) 
Après de nombreuses heures à essayer d'installer correctement 'javacv' dans le dossier 'libraries' de processing, je commence à douter de la nature du problème. Je n'ai habituellement aucune difficulté à installer manuellement une librairie dans p5 : pour faire fonctionner PapARt (en passant, bravo pour ce travail impressionnant !!) j'ai tenté de recréer différents path (javacv/ bytedeco/javacv etc) mais rien… En fait 'javacv' semble être bien pris en compte par p5 car ce dernier repère des doublons créés volontairement (dans cette optique) mais j'obtiens dans tous les cas le même message d'erreur :
**No library found for org.bytedeco.javacpp.opencv_core** 

Je me suis dit qu'il manquait peut-être un fichier .jar dans le dossier 'javacv' downloadable dans vos instructions ; je l'ai donc remplacé par l'archive binaire complète de bytedeco, mais même problème…
http://bytedeco.org/download/

Pour info, je peux lancer (en dépit du fameux message d'erreur) le sketch PCConfiguration.pde sans problème, tester la camera OpenCV sans faire planter p5, mais la camera Processing / FlyCapture (je voudrais utiliser dans un 1er temps la webcam simple de mon macbook pro) déclenchent un crash rédhibitoire du .pde (celui copié/collé dans mon message précédent).

Je loupe probablement un truc ultra évident ou alors le problème vient de ma version de p5 (3.3.6) ? Votre aide sera très précieuse : je voudrais utiliser PapArt avec des étudiants en design pour expérimenter de nouveaux types interfaces ;)
Merci !

**Comment by poqudrof** [comment]
Bonjour, 
Je n'ai pas trouvé non plus d'ou viennent ces messages d'erreur:  
```No library found for org.bytedeco.javacpp.opencv_core ```  
Cependant ils ne révèlent pas d'erreur et j'ai appris à les ignorer. Je vais chercher à la supprimer pour la version 1.1 mais rien n'est garanti car ils sont là depuis toujours :[ . 

Est-ce que la caméra fonctionne avec OpenCV ? Si c'est bon, pas besoin d'utiliser les drivers Processing. Pour utiliser les drivers Processing il faut recopier le nom de la caméra... Cette solution  est très mauvaise, et on pourrait la remplacer par une liste déroulante des caméras disponibles sur l'ordinateur.
Les drivers FlyCapture sont pour les caméras PointGreys (qui s'appellent FLIR maintenant, prix > 800€). 

Si la caméra fonctionne, tout est bon pour la suite. Les exemples de types `Camera` devraient fonctionner.

Quel est votre projet ? Je suis actuellement en train de réfléchir à monter un espace «communautaire» autour de la technologie. Je pourrais vous accompagner / supporter pour les développements à travers cet espace.

**Comment by Mameng** [comment]
Merci pour votre réponse.

Effectivement le message d'erreur en soit semble non bloquant et je comprends désormais les différentes modalités d'utilisation des caméras ! Effectivement la caméra OpenCV fonctionne et je vais donc pouvoir expérimenter votre libraire à partir de cela — mea culpa, je pensais que le "mode cam processing" était le mode par défaut …

Pour le reste je vous envoie de suite un message via votre adresse mail …
à bientôt !

**Comment by vantony1** [comment]
hello @poqudrof I am having the same issue -- I can't just ignore the error since nothing happens after I run any sketch in the get_started tutorial. 

I can't find support anywhere and I am trying to use the PapARt library for a undergrad research project. I would appreciate any help you could provide. thank yoU!

this is complete error I am getting

No library found for org.bytedeco.javacpp
Libraries must be installed in a folder named 'libraries' inside the sketchbook folder (see the Preferences window).
The package “org.bytedeco” does not exist. You might be missing a library.


This is how my libraries folder looks like.
![libs](https://user-images.githubusercontent.com/32144387/86979713-fed26580-c14f-11ea-87f1-d77d0c155bfd.PNG)

**Comment by poqudrof** [comment]
Hello, 

Everything seems quite OK, however you have to check that you have the correct binaries for JavaCV.  
Can you check that you have the windows x64 jar files in your Javacv/library directory ? 

JavaCPP is shipped with JavaCV, so that should be not be an issue directly.

**Comment by vantony1** [comment]
hey thank you for getting back to me! I really appreciate the help

I figured out the problem. I issue was that the jar files in my javacv folder had version numbers in their name so I just had to remove the version numbers from the filenames for processing to recognize them. 

I did bump into another problem however that my external pico projector connected through an hdmi cable is not being detected by the library even when I ran pccalibration. could help me figure out how to send output to a projector?

best,

vic
